core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java

```diff
@@ -958,11 +958,11 @@ public static Slice translate(@SqlType("varchar(x)") Slice source, @SqlType("var
 
     @Description("Encodes a string into a Soundex value")
     @ScalarFunction
-    @SqlType(StandardTypes.VARCHAR)
+    @SqlType("varchar(4)")
     public static Slice soundex(@SqlType(StandardTypes.VARCHAR) Slice slice)
     {
         try {
-            return utf8Slice(new Soundex().encode(slice.toStringUtf8()));
+            return utf8Slice(Soundex.US_ENGLISH.encode(slice.toStringUtf8()));
```

Comment:
static import

Reply:
I don't think that's a good idea. `US_ENGLISH` as static reference is too ambiguous.